core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -2994,8 +2994,14 @@ public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
       Object[] out = new Object[length];
 
       ExpressionType arrayType = null;
+
       for (int i = 0; i < length; i++) {
         ExprEval<?> evaluated = args.get(i).eval(bindings);
+        // short circuit the case where you have one input argument whose input type is array. So array function has
```

Comment:
i have a concern with this change, best illustrated by the implication that now an expression like this

```
array(array('x','y','z'))
```
will now produce `['x','y','z']` instead of `[['x','y','z']]` as one would expect, or would happen if the expression were this instead:

```
array(array('x','y','z'),array('a','b','c'))
```
which will still produce `[['x','y','z'],['a','b','c']]`

Instead of hijacking this function to accommodate multi-value string behavior, I think we should instead add a `MV_TO_ARRAY` function in SQL that only accepts a string, and then in the native expression layer either just use `cast(x, 'ARRAY<STRING>')` or also add a `mv_to_array` function that basically does the same thing as cast. This way we are explicit on when we intend to treat a multi-value column directly as an array

Reply:
The original intention was to keep it close to SQL but i realized SQL does not have multi value col's :). So a new function may be necessary . 
Cool I am adding a new `MV_TO_ARRAY` function to both the native and the SQL layer. 
Thanks for the tip

Reply:
Makes sense to me too. MV_TO_ARRAY would be simpler than ARRAY and have fewer cases to worry about. We can start there.